chipsec/modules/debugenabled.py

```diff
@@ -0,0 +1,120 @@
+#CHIPSEC: Platform Security Assessment Framework
+#Copyright (c) 2018, Eclypsium, Inc.
+# 
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; Version 2.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+
+"""
+This module checks if the system has debug features turned on, 
+specifiaclly the Direct Connect Interface (DCI).
+
+This module checks the following bits:
+1. HDCIEN bit in the DCI Control Register
+2. Debug enable bit in the IA32_DEBUG_INTERFACE MSR
+3. Debug lock bit in the IA32_DEBUG_INTERFACE MSR
+4. Debug occured bit in the IA32_DEBUG_INTERFACE MSR
+
+The module returns the following results:
+FAILED : Any one of the debug features is enabled or unlocked.
+PASSED : All debug feature are diabled and locked.
+
+Hardware registers used:
+IA32_DEBUG_INTERFACE[DEBUGENABLE]
+IA32_DEBUG_INTERFACE[DEBUGELOCK]
+IA32_DEBUG_INTERFACE[DEBUGEOCCURED]
+P2SB_DCI.DCI_CONTROL_REG[HDCIEN]
+
+"""
+
+from chipsec.module_common import *
+import chipsec.chipset
+import chipsec.defines
+_MODULE_NAME = 'debugenabled'
+
+
+import chipsec.hal.uefi
+import chipsec.hal.uefi_common
+
+EDX_ENABLE_STATE = 0x00000000
+IA32_DEBUG_INTERFACE_MSR = 0xC80
+P2SB_DCI_PORT_ID = 0xB8
+DCI_CONTROL_REG_OFFSET = 0x4
+
+HDCIEN_MASK = 0b00000000000000000000000000010000
+IA32_DEBUG_INTERFACE_DEBUGENABLE_MASK = 0b00000000000000000000000000000001
+IA32_DEBUG_INTERFACE_DEBUGELOCK_MASK = 0b01000000000000000000000000000000
+IA32_DEBUG_INTERFACE_DEBUGEOCCURED_MASK = 0b10000000000000000000000000000000 
+
+
+########################################################################################################
+#
+# Main module functionality
+#
+########################################################################################################
+
+class debugenabled(chipsec.module_common.BaseModule):
+
+
+
+    def __init__(self):
+        BaseModule.__init__(self)
+
+    def is_supported(self):
+        current_platform_id = self.cs.get_chipset_id()
+        supported = (current_platform_id == chipsec.chipset.CHIPSET_ID_CFL) | (current_platform_id == chipsec.chipset.CHIPSET_ID_KBL) | (current_platform_id == chipsec.chipset.CHIPSET_ID_SKL)
+        if not supported: self.logger.log_skipped_check( "DCI is not supported in this platform" )
+        return supported
```

Comment:
You are skipping the entire test if DCI is not supported.  Would you mind changing the is_supported function to just verify that the IA32_DEBUG_INTERFACE MSR exists and then check the platform before calling check_dci.  This gives us some better test coverage on older systems.

You can use CPU ID to determine if the IA32_DEBUG_INTERFACE MSR is supported on the platform using (CPUID (1,0) checking ecx bit 11).